src/components/Dashboard/CrossingStatusHistory/CrossingStatusHistoryFilter.js

```diff
@@ -0,0 +1,207 @@
+import React, { Component } from 'react';
+import { graphql } from 'react-apollo';
+import moment from 'moment';
+
+import ButtonSecondary from 'components/Shared/Button/ButtonSecondary';
+import ButtonPrimary from 'components/Shared/Button/ButtonPrimary';
+import Dropdown from 'components/Shared/Form/Dropdown';
+import DatePicker from 'components/Shared/DatePicker';
+import XIcon from 'components/Shared/Icons/XIcon';
+import { ALL_COMMUNITIES_NAME, ALL_COMMUNITIES_INDEX } from 'constants/AppConstants';
+import AllCommunitiesQuery from 'components/Shared/queries/AllCommunitiesQuery';
+
+class HistoryFilter extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      communityId: props.communityId,
+      dateLowerBound: props.dateLowerBound,
+      dateLowerBoundKey: new Date().toISOString(),
+      dateUpperBound: props.dateUpperBound,
+      dateUpperBoundKey: new Date().toISOString(),
+      inclusiveEndDate: props.inclusiveEndDate
+    }
+  }
+
+  clearLowerBound = () => {
+    this.setState({dateLowerBound: null});
+  }
+
+  clearUpperBound = () => {
+    this.setState({dateUpperBound: null});
+  }
+
+  dateLowerBoundChanged = (date) => {
+    this.setState({
+      dateLowerBound: date,
+      dateLowerBoundKey: new Date().toISOString(),
+    });
+  }
+
+  dateUpperBoundChanged = (date) => {
+    this.setState({
+      dateUpperBound: date,
+      dateUpperBoundKey: new Date().toISOString(),
+    });
+  }
+
+  inclusiveEndDateChanged = (value) => {
+    this.setState({inclusiveEndDate: value});
+  }
+
+  communityIdChanged = (e) => {
+    this.setState({communityId: Number(e.target.value)});
+  }
+
+  onSubmit = () => {
+    this.props.applyFilter({
+      communityId: this.state.communityId,
+      dateLowerBound: this.state.dateLowerBound,
+      dateUpperBound: this.state.dateUpperBound,
+      inclusiveEndDate: this.state.inclusiveEndDate
+    });
+  }
+
+  onCancel = () => {
+    this.setState((state, props) => {
+      return {
+        communityId: props.communityId,
+        dateLowerBound: props.dateLowerBound,
+        dateUpperBound: props.dateUpperBound,
+        inclusiveEndDate: props.inclusiveEndDate
+      }
+    })
+  }
+
+  render() {
+    const {
+      communityId,
+      dateLowerBound,
+      dateLowerBoundKey,
+      dateUpperBound,
+      dateUpperBoundKey,
+      inclusiveEndDate,
+    } = this.state;
+
+    let allCommunities;
+    const communitiesLoading = !this.props.AllCommunitiesQuery || this.props.AllCommunitiesQuery.loading;
+    if (communitiesLoading) {
+      return <div>Loading...</div>
+    }
+    allCommunities = this.props.AllCommunitiesQuery.allCommunities.nodes.filter(node => node.id !== 1337 && node.id !== 9002);
```

Comment:
not the biggest fan of having 1337 and 9002 in here as magic numbers but it def makes sense for the moment. Worth noting that we're using them here at least.

Reply:
Yeah there's one other place where I'm doing that too. It's a temporary measure until I figure out a plan for Super_Admin_World and ALL.